src/format/opamFile.ml

```diff
@@ -541,7 +592,7 @@ end
 
 (** Cached environment updates (<switch>/.opam-switch/environment) *)
 
-module Environment = LineFile(struct
+module Environment = AtomicLineFile(struct
```

Comment:
It is better to split this commit into 2 : one that introduce atomic writing, the other that change environment writing.

Reply:
Will do - in fact, I think it's worth splitting the atomic writing part off completely

Reply:
I extracted the commit in #5473